### PR TITLE
Monitoring metrics and additional logging

### PR DESCRIPTION
**Pull request by ssd04**
Added more expliocit logging for rabbitmq and redis integrations.
Monitoring metrics for redis and rabbitmq operations.

### CODE REVIEW

File: metrics/statusMetrics_test.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	for i := 0; i < numIterations; i++ {
+		go func(index int) {
+			switch index % 3 {
+			case 0:
+				sm.AddRequest(fmt.Sprintf("op_%d", index%5), time.Hour*time.Duration(index))
+			case 1:
+				res := sm.GetAll()
+				delete(res, "op_0")
+			case 2:
+				_ = sm.GetMetricsForPrometheus()
```
Comment (bogdan-rosianu) [inline metrics/statusMetrics_test.go:141]: might call the `GetAll` as well

File: api/groups/statusGroup.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+// getMetrics will expose endpoints statistics in json format
+func (sg *statusGroup) getMetrics(c *gin.Context) {
+	metricsResults := sg.facade.GetMetrics()
+
+	shared.JSONResponse(c, http.StatusOK, gin.H{"metrics": metricsResults}, "")
+}
+
+// getPrometheusMetrics will expose proxy metrics in prometheus format
```
[the remark this answers is not in the document]
Comment (ssd04) [reply api/groups/statusGroup.go:62]: :+1:

File: api/groups/statusGroup.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+			Handler: sg.getPrometheusMetrics,
+			Method:  http.MethodGet,
+		},
+	}
+	sg.endpoints = endpoints
+
+	return sg, nil
+}
+
+// getMetrics will expose endpoints statistics in json format
```
[the remark this answers is not in the document]
Comment (ssd04) [reply api/groups/statusGroup.go:55]: changed

File: metrics/statusMetrics_test.go (hunk 1 above)
Comment (ssd04) [reply metrics/statusMetrics_test.go:141]: updated test for case 1

File: rabbitmq/rabbitClient.go [hunk 4]
```diff
@@ -78,13 +78,16 @@ func (rc *rabbitMqClient) Publish(exchange, key string, mandatory, immediate boo
 			immediate,
 			msg,
 		)
+		if err != nil {
+			log.Error("failed to publish message", "exchange", exchange, "error", err.Error())
```
[the remark this answers is not in the document]
Comment (ssd04) [reply rabbitmq/rabbitClient.go:82]: key field is a bit misleading, it is not the payload key and not actually used, it's set to empty string

File: redis/connection.go [hunk 5]
```diff
@@ -18,6 +18,8 @@ func CreateSimpleClient(cfg config.RedisConfig) (RedLockClient, error) {
 	}
 	client := redis.NewClient(opt)
 
+	log.Debug("created redis instance connection type", "connection url", cfg.Url, "master")
```
[the remark this answers is not in the document]
Comment (ssd04) [reply redis/connection.go:21]: right, master is not used for single instance connection, removed "master" key

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by miiu96 — COMMENTED** [review]
maybe you can extract all the API endpoints in a config file, this can be done in a future PR

**Review by miiu96 — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**